internal/internal_workflow_client.go

```diff
@@ -190,7 +190,8 @@ func (wc *workflowClient) StartWorkflow(
 	}
 
 	if wc.metricsScope != nil {
-		wc.metricsScope.Counter(metrics.WorkflowStartCounter).Inc(1)
+		tagScope(wc.metricsScope, tagWorkflowType, workflowType.Name).
```

Comment:
this will only include that tag for this specific metric. We want to include those tags for all metrics. But if you do the same thing for every metric, you are going to create a log of tagged subscope, which is not a cheap operation.